language/evm/move-to-yul/src/native_functions.rs

```diff
@@ -39,15 +39,12 @@ impl NativeFunctions {
             emit!(ctx.writer, "function {}", fun_name);
             ngen(gen, ctx, fun_id)
         } else {
-            ctx.env.error(
-                &gen.contract_loc,
-                &format!(
-                    "native function `{}` not implemented for type `{:?}`",
-                    ctx.env
-                        .get_function(fun_id.to_qualified_id())
-                        .get_full_name_str(),
-                    fun_id.inst,
-                ),
+            unimplemented!(
```

Comment:
There is a simple rule which one can apply here: if a user input (or any external input) into a program can lead to an error situation, the program should never crash, but gracefully produce an error. Since a user can just declare any kind of native functions, we should better continue to produce an error.

Reply:
> Since a user can just declare any kind of native functions, we should better continue to produce an error.

I see your point and I agree. Now I think this actually reveals a bigger problem: right now, the native function implementations are part of the move-to-yul compiler itself, so it's not something the caller can control, leading to confusion. I think we should
1. move native function implementations into move-stdlib (preferrable gated behind a feature flag)
2. define a native function interface for the move-to-yul compiler and let the caller pass implementations in, similar to how MoveVM handles its native function implementations

cc @emmazzz

Reply:
I think it might be sufficient to have an error message like: "unexpected undefined native function (bug?)"? At least for now.